ros2bag/ros2bag/verb/record.py

```diff
@@ -122,11 +130,17 @@ def add_arguments(self, parser, cli_name):  # noqa: D102
 
     def main(self, *, args):  # noqa: D102
         # both all and topics cannot be true
-        if args.all and args.topics:
-            return print_error('Invalid choice: Can not specify topics and -a at the same time.')
-        # both all and topics cannot be false
-        if not(args.all or (args.topics and len(args.topics) > 0)):
-            return print_error('Invalid choice: Must specify topic(s) or -a')
+        if (args.all and (args.topics or args.regex)) or (args.topics and args.regex):
+            return print_error('Must specify only one option out of topics, regex or -a')
```

Comment:
In help messages, please use long-form versions for clarity - power users will always find or know the short form

```suggestion
            return print_error('Must specify only one option out of topics, --regex or --all')
```